### PR TITLE
Declare version numbers and install cmake version config file

### PR DESCRIPTION
**Pull request by scpeters**
The cmake version config file is similar to the version added in https://github.com/ros/urdfdom_headers/pull/17 and the logic for hte Windows install location is based on https://github.com/ros/urdfdom/pull/41 by @traversaro

Some of the open issues call for version number bumping, so this should help with that.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,7 +51,26 @@ set(PKG_NAME ${PROJECT_NAME})
...
+include(WriteBasicConfigVersionFile)
+write_basic_config_version_file(
+  ${CMAKE_CURRENT_BINARY_DIR}/${cmake_conf_version_file}
+  VERSION "${CONSOLE_BRIDGE_VERSION}"
+  COMPATIBILITY SameMajorVersion)
+
+if(WIN32 AND NOT CYGWIN)
+  set(CMAKE_CONFIG_INSTALL_DIR CMake)
+else()
+  set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/cmake/)
```
Comment (dirk-thomas) [inline CMakeLists.txt]: Out of curiosity: where does `CMAKE_INSTALL_DATAROOTDIR` point to? And where does it get set?

Comment (scpeters) [inline CMakeLists.txt]: whoops, that is set by `GNUInstallDirs`

Comment (scpeters) [inline CMakeLists.txt]: which was added in https://github.com/ros/urdfdom/pull/41 but not here

Comment (scpeters) [inline CMakeLists.txt]: I was just trying to be consistent with that other pull request, though I know there has been some controversy about GNUInstallDirs

Comment (scpeters) [inline CMakeLists.txt]: To be added in https://github.com/ros/console_bridge/pull/23

**Pull request merged**